pkg/querysharding/analyzer.go

```diff
@@ -55,15 +57,16 @@ func (a *QueryAnalyzer) Analyze(query string) (QueryAnalysis, error) {
 			}
 		case *parser.BinaryExpr:
 			if n.VectorMatching != nil {
-				analysis = analysis.scopeToLabels(n.VectorMatching.MatchingLabels, n.VectorMatching.On)
+				//Exclude le label from sharding
+				shardingLabels = excludeLabel(n.VectorMatching.MatchingLabels, "le")
```

Comment:
I don't we should not introduce the new `excludeLabels` function, and we should reuse the existing methods that we have in the `analysis` struct.

Reply:
Yeah didn't realize we already have `without`. Will replace.